util/util.go

```diff
@@ -94,6 +96,50 @@ func GetServiceIDByName(c rest.FrontClient, name string) (akid.ServiceID, error)
 	return akid.ServiceID{}, errors.Errorf("cannot determine project ID for %s", name)
 }
 
+func VerifyServiceByServiceID(c rest.FrontClient, serviceID akid.ServiceID) error {
+	unexpectedErrMsg := "Something went wrong while starting the Agent. " +
+		"Please contact Postman support (observability-support@postman.com) with the error details"
+	failedToVerifyServiceErrMsg := "Failed to verify service for given serviceID: %s\n"
+
+	// Check if service is already verified and cached
+	if service, found := serviceIDCache.Get(akid.String(serviceID)); found {
+		printer.Stderr.Debugf("Cached service %v for serviceID %s\n", service, akid.String(serviceID))
+		return nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), apiTimeout)
+	defer cancel()
+
+	service, err := c.GetService(ctx, serviceID)
+	if err != nil {
+		httpErr, ok := err.(rest.HTTPError)
+		if !ok {
+			printer.Stderr.Debugf(failedToVerifyServiceErrMsg, err)
+			return errors.Wrap(err, unexpectedErrMsg)
+		}
+
+		var errorResponse rest.ErrorResponse
+		if err := json.Unmarshal(httpErr.Body, &errorResponse); err != nil {
+			printer.Stderr.Debugf(failedToVerifyServiceErrMsg, err)
+			return errors.Wrap(err, unexpectedErrMsg)
+		}
+
+		if httpErr.StatusCode == 404 {
+			return fmt.Errorf("there is not service with given ID %s. Ensure that your service ID is correct", serviceID)
+		} else if httpErr.StatusCode == 403 {
+			return fmt.Errorf("you cannot send traffic to the service with ID %s. "+
```

Comment:
Blocking: It's not clear to me without digging deeper how these will be displayed.  However, it seems inconsistent that the errors messages on line 100 and 102 are capitalized, while these two are not.  So we should resolve that to ensure the error is displayed in a way that looks good to the user.

```suggestion
			return fmt.Errorf("There is no service with given ID %s. Ensure that your service ID is correct", serviceID)
		} else if httpErr.StatusCode == 403 {
			return fmt.Errorf("You cannot send traffic to the service with ID %s. "+
```

I don't think "service" and "service ID" are the terms we are using, can you confirm with our documentation of what we are calling the Insights projects?  In user-facing messages we should ensure we're following consistent naming -- even though it will be some work to work through all the places where we say "services" in the CLI right now, let's not introduce new ones.

Reply:
> it seems inconsistent that the errors messages on line 100 and 102 are capitalized, while these two are not

Have not capitalized them due to the lint warning, added lint ignore.